### ISSUE TITLE
[1610.02242] Temporal Ensembling for Semi-Supervised Learning

### ISSUE THREAD

**Issue by riron1206** [riron1206/read_paper#18]
## 概要
https://arxiv.org/abs/1610.02242
Π-model、Temporal Ensemble を提案した論文
画像変化加えても出力ラベルは一貫(consistency regularization)するという考えにもとづいた半教師学習
（2019年に提案されたMixMatchでもconsistency regularizationの考え導入してる）
![Π-model](https://user-images.githubusercontent.com/10371630/41501166-caf8a93c-7196-11e8-9072-19dd6968c8e6.png)

## 新規性・差分
consistency regularization

## 手法
■Π-model
data augmentation実行してdropout入れたネットワーク
lossがcross entropy(教師ありのロス)とmean squared error(教師なしのロス)の合計
loss = L(cross entropy) + w(epoch)*L(mse)
教師なしのロスの強さをw(epoch)で調節する
- cross entropyはラベルありサンプルにのみ適用
- mean squared errorはラベルなしサンプルの予測ラベルについて適用。同じラベルなしサンプルをモデルに2回流し、データの拡張とドロップアウトによって変化した2データについて誤差をとる

■Temporal ensembling
各epochの予測ラベルの平均値を保持するように改良したΠ-model
同じラベルなしサンプルについて、1回だけモデルに流し、これまでのepochの予測ラベルの平均値とで誤差をとる
1回だけ流すから学習速度短縮できる
サンプルごとに予測ラベル持ち続けないとダメだからメモリめっちゃいる
![Π-model]()

## 結果
CIFAR-10などでこれまでの半教師の方法よりerror率下げた

## コメント
参考サイト
- https://www.cnblogs.com/wuliytTaotao/p/12825797.html
- http://37ma5ras.blogspot.com/2018/06/2017-temporal-ensembling-for-semi.html
- https://tech-blog.abeja.asia/entry/ml-annotation
- https://hotcocoastudy.hatenablog.jp/entry/2020/01/23/210357
